src/internal.c

```diff
@@ -16200,7 +16196,7 @@ int ProcessReply(WOLFSSL* ssl)
         atomicUser = 1;
 #endif
 
-    if (ssl->error != 0 && ssl->error != WANT_READ && ssl->error != WANT_WRITE
+    if (checkError == 1 && ssl->error != 0 && ssl->error != WANT_READ && ssl->error != WANT_WRITE
```

Comment:
In the event of an alert, `ssl->error != 0`, so we need to not do this check when we're calling `ProcessReply` specifically for the purpose of reading the alert message.

Reply:
I'm thinking that on an error from a Send* function, we should only do a `ProcessReplyEx` on a socket error. Otherwise we are ignoring the `ProcessReplyEx` return.

Reply:
Yeah, I think only whitelisting socket errors for this use case makes sense.